### ISSUE TITLE
Issues with List headers, List footers, and List entry totals

### ISSUE THREAD

**Issue by KentSpiel** [usfm-bible/tcdocs#20]
The current Documentation does not describe the Biblica use case for these markers.
### List header
[The current USFM 3.0 documentation for \lh](https://ubsicap.github.io/usfm/lists/index.html#lh) states:
>Some lists include an introductory and concluding remark ([\lf](https://ubsicap.github.io/usfm/lists/index.html#usfmp-lf)). They are an integral part of the list content, but are not list items. A list does not require either or both of these elements.

Although these are not list items they are list elements. Biblica calls them List Introductions and List Conclusions, formats them using list indents, and outdents verse numbers to show their integration with the list.
This example from EXO 6:14 illustrates a List Introduction which we mark as  \lh followed by a \b

 ![List Header from NIV EXO6 14](https://user-images.githubusercontent.com/6817239/183447035-6e664725-dc75-4c39-9b75-b6761be92b2b.png)

This example from EZR 2:55 illustrates a Sublist Introduction which we mark as \lh. It is not followed by a \b [**Note:** replace this  image from an older typeset with one from a newer typeset that does not have a space following.]
![Sublist Heading from NIV EZR2 55](https://user-images.githubusercontent.com/6817239/183456317-ad3b9a53-d58e-4a6a-b010-8d39b0470c42.png)

_I would like an illustration from the NIV to be used to illustrate proper and allowable uses and formatting of `\lh`_ 

### List footer
[The current Documentation for \lf](https://ubsicap.github.io/usfm/lists/index.html#lh) states:
>Some lists include an introductory ([\lh](https://ubsicap.github.io/usfm/lists/index.html#usfmp-lh)) and concluding remark. They are an integral part of the list content, but are not list items. A list does not require either or both of these elements.

The current USFM 3.0.4 does not support `\lf` with `\litl ...\litl*`. We need to use `\lf` as a Total line. [Appropriate illustration from EZR 2:55 to be added.]  I believe all list types should allow List Totals including Headings. (A translation may want to put the total first.)

_I would like the `\OccursUnder` attribute of ` \Marker litl` to include `\lh` and `\lf`_

### List entry total
- The current documentation shows this markup:
```
\lim1
\v 11 of Pahath-Moab (through the line of Jeshua and Joab) - \litl 2,818\litl*
\lim1
\v 12 of Elam - \litl 1,254\litl*
```
The hyphen disappears and is replaced by dotted leader.
![image](https://user-images.githubusercontent.com/6817239/183473989-15d0e4cd-593e-4c1f-9752-ea99b6a6c2b9.png)

It is unclear from the documentation why that is. Some of out projects use dotted leader (Russian) and others do not (English). I am currently using three leader dots in place of the hyphen to represent a dotted leader. I am using there tildes  ~~~ to represent no leader dots. This formats reasonably well in Paratext, but I do not think that will communicate to partners using the USX representation for publication.)

_I would like a formal way of specifying the preferred leader type for a **List entry total**._

**Comment by davidg-sil** [comment]
My gut feeling is that:
a) since USFM is (ideally) semantic markup, the leader type should be defined in some out-of-band configuration.
b) If there are different classes of lists (and tables) then `\cat somecategory\cat*` on the first `\tr` or `\li` line ought to be used to specify it.
c) there *really* shouldn't be any magic characters that are (sometimes?) supposed to vanish when followed by a `\litl` marker. It seems too much like magic.

**Comment by mhosken** [comment]
I agree that coming up with a special encoding standard on top of USFM/X to address what is purely a styling matter and not semantic at all, is not a good idea. The hyphen in the example is just that: a hyphen. It says nothing more.

What is problematic about USFM/X is that the highest grouping it has is the paragraph. A USX file is a sequence of paragraphs (basically). Even sidebars are done using a kind of milestone (in USFM. They are an identified unit in USX). But there is nothing that provides for a grouping of paragraphs in the main text. It might be tempting to add something. But I think the best approach is to use a pair of milestones. Thus I would be up for defining a couple of milestones that allow setting the category for the paragraphs within (much like what happens with a sidebar, only not).

I'm even in favour of adding <cat> to paragraphs. But that isn't necessary for this, where what you really want is to set a category (to indicate the category of typesetting to use, not the concrete question of leaders or not) for the whole list. Hence I would suggest a milestone with a category attribute.

```
\div-s|cat="text"\*
```

This is a list in the main text hence it is a text list rather than say an introduction list or whatever.
